gems/sorbet-runtime/lib/types/private/methods/call_validation.rb

```diff
@@ -123,7 +123,14 @@ def self.validate_call(instance, original_method, method_sig, args, blk)
     # this code is sig validation code.
     # Please issue `finish` to step out of it
 
-    return_value = original_method.bind(instance).call(*args, &blk)
+    bound_method = original_method.bind(instance)
+    return_value =
+      if original_method.parameters.any? { |param| param.first == :keyreq }
```

Comment:
Could this check be done _ahead_ of time?
E.g. let's have two versions of this method, one where `original_method.parameters.any? { |param| param.first == :keyreq }` is true and another where its false.

This should also fix failing test, right?

Reply:
That's an interesting idea. Since you think the broad concept has merit, I'll put more time into it and see what I can come up with.